### PR TITLE
AITOOL 4699 - Upgraded to 2.0.0-beta.1

### PR DESCRIPTION
**Pull request by joseangel-mm**
# Jira ticket: [AITOOL-4699](https://lampkicking.atlassian.net/browse/AITOOL-4699)

## Purpose
Upgrade the FCM to 2.0.0-beta.1.

[AITOOL-4699]: https://lampkicking.atlassian.net/browse/AITOOL-4699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -118,6 +116,8 @@ const App = () => {
                 onSuccess={onSuccess}
                 onError={onError}
                 secure={secureFlag}
+                clientSdkId={""}
```
Comment (tonygcs) [inline src/App.js:119]: Do you think it is worth using a webpack plugin to get the value from `process.env.SDK_ID` for the `clientSdkId`?

File: README.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ Both `PEM_FILE_PATH` and `SDK_ID` secrets are obtained from following the instru
 
 The information required to fill the `ENDPOINT` variable can be found [here](https://developers.yoti.com/age-estimation/integration-guide). The `BASE_URL` (base URL for the service to be requested) is set in the `.env` file (check the value to fill the `ENDPOINT` field).
 
+In the `clientSdkId` field in the [FaceCapture component](https://github.com/getyoti/web-fcm-demo/blob/04738a2a43258594872c28b6bced669c6789ad49/src/App.js#L117) you must set the same `sdk_id` as the one in the env.local.
```
Comment (tonygcs) [inline README.md:25]: I would rephrase this sentence with something like:

```
Set the `clientSdkId` in the [FaceCapture component](src/App.js#L117) with `SDK_ID` value in the `.env.local` file.
```

What do you think?

File: src/App.js [hunk 3]
```diff
@@ -73,12 +72,11 @@ const App = () => {
 
   const classes = useStyles();
 
-  const onSuccess = ({ img, secure }) => {
-    setImage(img);
-    const fcmResponse = { img, secure };
```
[the remark this answers is not in the document]
Comment (tonygcs) [reply src/App.js:78]: The `onSuccess` callback already returns the `payload` that contains `secure` in case the mode is enabled and it is used [here](https://github.com/getyoti/web-fcm-demo/pull/30/files#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67R79) to build the final request so the `fcmResponse` variable is not needed anymore.

[the remark this answers is not in the document]
Comment (pegondo) [reply src/App.js:78]: Nothing actually changed here, the object was unwrapped before and then wrapped, and now it is never unwrapped.

Previous `fcmResponse` === new `payload`

**Review by pegondo — APPROVED** [review]

File: src/App.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (MiguelJS) [reply src/App.js:78]: thanks for the clarification

File: src/App.js (hunk 1 above)
Comment (joseangel-mm) [reply src/App.js]: Sure, already got the value from the `.env.local`.

**Review by MiguelJS — APPROVED** [review]

**Review by DionisioRD — APPROVED** [review]
LGTM Is good to have the branch usable, but don't merge it. We will merge to main once the final release of 2.0.0 is released and implemented.

File: webpack.config.js [hunk 4]
```diff
@@ -2,6 +2,15 @@ const HtmlWebPackPlugin = require("html-webpack-plugin");
...
 const path = require("path");
 
+const webpack = require('webpack');
+const dotenv = require('dotenv');
+
+const env = dotenv.config({path: '.env.local'}).parsed;
+const envKeys = Object.keys(env).reduce((prev, next) => {
+  prev[`process.env.${next}`] = JSON.stringify(env[next]);
+  return prev;
+}, {});
```
Comment (tonygcs) [inline webpack.config.js:12]: I was wondering if we can remove these steps that handle specific `.env` files and I had a look into the webpack documentation. They recommend the usage of [dotenv-webpack](https://webpack.js.org/plugins/environment-plugin/#dotenvplugin). Do you think that we could use it?

**Comment by tonygcs** [comment]
> LGTM Is good to have the branch usable, but don't merge it. We will merge to main once the final release of 2.0.0 is released and implemented.

Exactly, could you point the PR to the branch `2.0.0-beta.1`, please?

File: webpack.config.js (hunk 4 above)
Comment (joseangel-mm) [reply webpack.config.js]: Nice suggestion. I've changed the library and now it looks better, thanks.

**Comment by joseangel-mm** [comment]
> > LGTM Is good to have the branch usable, but don't merge it. We will merge to main once the final release of 2.0.0 is released and implemented.
> 
> Exactly, could you point the PR to the branch `2.0.0-beta.1`, please?

This is done.

**Review by tonygcs — APPROVED** [review]

File: webpack.config.js (hunk 4 above)
Comment (tonygcs) [reply webpack.config.js]: Thanks!!!!

**Review by MiguelJS — APPROVED** [review]
